### PR TITLE
First pass at an auditing plugin for hapi

### PR DESCRIPTION
**Pull request by christopherjturner**

### CODE REVIEW

File: src/config/config.js [hunk 1]
```diff
@@ -221,6 +221,18 @@ const config = convict({
     nullable: true,
     default: null,
     env: 'CDP_HTTPS_PROXY'
+  },
+  auditingEnabled: {
+    doc: 'Enable sending audit events to the CDP auditing stack.',
+    format: Boolean,
+    default: process.env.NODE_ENV === 'production',
```
Comment (flurdy) [inline src/config/config.js:228]: Will this also fire in a docker compose environment?

Comment (flurdy) [reply src/config/config.js:228]: Do we need to enable firehose on localstack?

File: src/server/common/helpers/audit/impl/aws-auditor.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import { config } from '~/src/config/config'
+import { FirehoseClient, PutRecordCommand } from '@aws-sdk/client-firehose'
+import { auditSchema } from '~/src/server/common/helpers/audit/schema/audit-schema'
+import { createLogger } from '~/src/server/common/helpers/logging/logger'
+import { fromNodeProviderChain } from '@aws-sdk/credential-providers'
+
+class AwsAuditor {
+  static _firehose
+  _source = 'cdp-portal-frontend'
```
Comment (flurdy) [inline src/server/common/helpers/audit/impl/aws-auditor.js:9]: Can this be with config?

File: src/config/config.js (hunk 1 above)
Comment (christopherjturner) [reply src/config/config.js:228]: yeap, tested it locally, we just need to override the endpoint on the client for it to work

File: src/server/common/helpers/audit/impl/aws-auditor.js (hunk 2 above)
Comment (christopherjturner) [reply src/server/common/helpers/audit/impl/aws-auditor.js]: done

**Pull request merged**